lib/bolt/cli.rb

```diff
@@ -426,58 +426,72 @@ def execute(options)
-                           targets,
-                           options[:task_options],
-                           executor) do |event|
-                outputter.print_event(event)
-              end
-            when 'file'
-              src = options[:object]
-              dest = options[:leftovers].first
+      handler = Signal.trap :INT do |signo|
+        @logger.info([
+          "Exiting after receiving SIG#{Signal.signame(signo)} signal.",
+          'There may be processes left executing on some nodes.'
+        ].join(' '))
+        exit!
+      end
 
-              if dest.nil?
-                raise Bolt::CLIError, "A destination path must be specified"
+      begin
+        if options[:mode] == 'plan'
+          executor = Bolt::Executor.new(@config, options[:noop], true)
+          result = pal.run_plan(options[:object], options[:task_options], executor)
+          outputter.print_plan_result(result)
+          # An exception would have been raised if the plan failed
+          code = 0
+        else
+          executor = Bolt::Executor.new(@config, options[:noop])
+          targets = options[:nodes]
+
+          results = nil
+          outputter.print_head
+
+          elapsed_time = Benchmark.realtime do
+            results =
+              case options[:mode]
+              when 'command'
+                executor.run_command(targets, options[:object]) do |event|
+                  outputter.print_event(event)
+                end
+              when 'script'
+                script = options[:object]
+                validate_file('script', script)
+                executor.run_script(
+                  targets, script, options[:leftovers]
+                ) do |event|
+                  outputter.print_event(event)
+                end
+              when 'task'
+                pal.run_task(options[:object],
+                             targets,
+                             options[:task_options],
+                             executor) do |event|
+                  outputter.print_event(event)
+                end
+              when 'file'
+                src = options[:object]
+                dest = options[:leftovers].first
+
+                if dest.nil?
+                  raise Bolt::CLIError, "A destination path must be specified"
+                end
+                validate_file('source file', src)
+                executor.file_upload(targets, src, dest) do |event|
+                  outputter.print_event(event)
+                end
               end
-              validate_file('source file', src)
-              executor.file_upload(targets, src, dest) do |event|
-                outputter.print_event(event)
-              end
-            end
-        end
+          end
 
-        outputter.print_summary(results, elapsed_time)
-        code = results.ok ? 0 : 2
+          outputter.print_summary(results, elapsed_time)
+          code = results.ok ? 0 : 2
+        end
+      ensure
```

Comment:
I think this can just be an ensure on the method
```
ensure
  Signal.trap :INT, 'DEFAULT'
```

Reply:
What will that do when running tests through rspec which sets its own signal handlers?

Reply:
I guess that question already sort of applies regardless, since we're setting up a signal handler that calls `exit!`...

Reply:
yeah, that should work